Created streaming feature for AioHttp_OpenAI

Kashyab19 opened the pull request:
## Title

Adding Streaming Feature for AIOHTTP_OPENAI

<!-- e.g. "Implement user authentication feature" -->

## Relevant issues

<!-- e.g. "Fixes #000" -->


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature

## Changes

<!-- List of changes -->
- Added ge_model

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locally
If UI changes, send a screenshot/GIF of working UI fixes

<!-- Test procedure -->

Commit:
Created streaming feature for AioHttp_OpenAI

ishaan-jaff reviewed (changes requested):
looks good in general - can you add testing and make sure you're testing passes 

add here `test_aiohttp_openai.py` 

Add the following test and pass the scenario - the function `streaming_format_tests` is defined in `test_streaming.py` 

```python
def test_completion_model_stream(model):
    try:
        messages = [
            {"role": "system", "content": "You are a helpful assistant."},
            {
                "role": "user",
                "content": "how does a court case get to the Supreme Court?",
            },
        ]
        response = completion(
            model=model, messages=messages, stream=True, max_tokens=50
        )
        complete_response = ""
        # Add any assertions here to check the response
        for idx, chunk in enumerate(response):
            chunk, finished = streaming_format_tests(idx, chunk)
            if finished:
                break
            complete_response += chunk
        if complete_response.strip() == "":
            raise Exception("Empty response received")
        print(f"completion_response: {complete_response}")
    except Exception as e:
        pytest.fail(f"Error occurred: {e}")

```

ishaan-jaff commented on litellm/llms/aiohttp_openai/common_utils.py:
```diff
@@ -0,0 +1,137 @@
+    def __iter__(self):
+        return self
+
+    def __next__(self):
```
this is not needed, you can just inherit this from `BaseModelResponseIterator` 


example iterator in cloudflare/chat/transformation.py

```python
class CloudflareChatResponseIterator(BaseModelResponseIterator):
    def chunk_parser(self, chunk: dict) -> GenericStreamingChunk:
        try:
            text = ""
            tool_use: Optional[ChatCompletionToolCallChunk] = None
            is_finished = False
            finish_reason = ""
            usage: Optional[ChatCompletionUsageBlock] = None
            provider_specific_fields = None

            index = int(chunk.get("index", 0))

            if "response" in chunk:
                text = chunk["response"]

            returned_chunk = GenericStreamingChunk(
                text=text,
                tool_use=tool_use,
                is_finished=is_finished,
                finish_reason=finish_reason,
                usage=usage,
                index=index,
                provider_specific_fields=provider_specific_fields,
            )

            return returned_chunk

        except json.JSONDecodeError:
            raise ValueError(f"Failed to decode JSON from chunk: {chunk}")

```

ishaan-jaff reviewed (commented):
-

krrishdholakia commented:
@Kashyab19 can you add testing for this PR

Here's an example - https://github.com/BerriAI/litellm/blob/a7b3c664d1b05dc6289f956e53b455f40d671bb3/tests/local_testing/test_streaming.py#L677

Commit:
Feature: Streaming for OpenAI via aiohttp
- Handled streaming via async calls in the handler
- Processed the chunks according the OpenAI Response Format
- Wrote Test Case for Streaming

Kashyab19 commented:
# Test Cases Result:
![AIOHTTP_STREAM_TEST](https://github.com/user-attachments/assets/e6c89bb9-848a-4320-b311-f582b9d9fd27)

ishaan-jaff reviewed (approved):
lgtm
